engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -1217,6 +1217,7 @@ paladin_td_t::paladin_td_t( player_t* target, paladin_t* paladin ) :
   debuff.judgment_of_light = make_buff( *this, "judgment_of_light", paladin -> find_spell( 196941 ) );
   debuff.final_reckoning = make_buff( *this, "final_reckoning", paladin -> talents.final_reckoning );
   debuff.reckoning = make_buff( *this, "reckoning", paladin -> spells.reckoning );
+  debuff.vengeful_shock = make_buff( *this, "vengeful_shock", paladin -> find_spell( 340007 ) );
```

Comment:
Conduits don't change value throughout combat, you should be able to just set the default value here rather than calling it on every trigger.

Additionally, you should be able to replace the find_spell() with conduit.vengeful_shock->effectN( 1 ).trigger()